### PR TITLE
PBM-345 Centralize logs

### PR DESCRIPTION
**Pull request by dAdAbird**
Added logs listing into pbm CLI

`pbm logs` options:
```
  --tail=20                  Show last N entries
  --node=NODE                Target node in format replset[/host:posrt]
  --type=TYPE                Entry type <INFO>/<Warning>/<ERROR>
  --event=EVENT              Event in format backup[/2020-10-06T11:45:14Z]
```

Requests & output examples:
```
$ pbm logs --tail=100 --node=rs1/localhost:37020 --event=backup
2020-10-15T20:42:07.000+0300 [INFO] backup/2020-10-15T17:41:50Z: backup started
2020-10-15T20:42:10.000+0300 [INFO] backup/2020-10-15T17:41:50Z: s3.uploadPartSize is set to 10485760 (~10Mb)
2020-10-15T20:42:10.000+0300 [ERROR] backup/2020-10-15T17:41:50Z: backup: mongodump: write data: upload to S3: RequestError: send request failed
caused by: Put "http://127.0.0.1:9000/bkt/pbm/2020-10-15T17%3A41%3A50Z_rs1.dump.s2": dial tcp 127.0.0.1:9000: connect: connection refused.
2020-10-15T20:43:11.000+0300 [INFO] backup/2020-10-15T17:42:54Z: backup started
2020-10-15T20:43:14.000+0300 [INFO] backup/2020-10-15T17:42:54Z: s3.uploadPartSize is set to 10485760 (~10Mb)
2020-10-15T20:43:14.000+0300 [INFO] backup/2020-10-15T17:42:54Z: mongodump finished, waiting for the oplog
2020-10-15T20:43:43.000+0300 [INFO] backup/2020-10-15T17:42:54Z: s3.uploadPartSize is set to 10485760 (~10Mb)
2020-10-15T20:43:48.000+0300 [INFO] backup/2020-10-15T17:42:54Z: backup finished
```
```
$ pbm logs --tail=100 --node=rs1/localhost:37020 --event=backup --type=ERROR
2020-10-15T20:42:10.000+0300 [ERROR] backup/2020-10-15T17:41:50Z: backup: mongodump: write data: upload to S3: RequestError: send request failed
caused by: Put "http://127.0.0.1:9000/bkt/pbm/2020-10-15T17%3A41%3A50Z_rs1.dump.s2": dial tcp 127.0.0.1:9000: connect: connection refused.
```
```
$ pbm logs --tail=100 --node=rs1 --event=backup --type=ERROR
2020-10-15T20:42:10.000+0300 rs1/localhost:37020 [ERROR] backup/2020-10-15T17:41:50Z: backup: mongodump: write data: upload to S3: RequestError: send request failed
caused by: Put "http://127.0.0.1:9000/bkt/pbm/2020-10-15T17%3A41%3A50Z_rs1.dump.s2": dial tcp 127.0.0.1:9000: connect: connection refused.
```
```
$ pbm logs --tail=100 --event=backup --type=ERROR
2020-10-15T20:42:10.000+0300 rs1/localhost:37020 [ERROR] backup/2020-10-15T17:41:50Z: backup: mongodump: write data: upload to S3: RequestError: send request failed
caused by: Put "http://127.0.0.1:9000/bkt/pbm/2020-10-15T17%3A41%3A50Z_rs1.dump.s2": dial tcp 127.0.0.1:9000: connect: connection refused.
2020-10-15T20:42:10.000+0300 rs2/localhost:28020 [ERROR] backup/2020-10-15T17:41:50Z: backup: mongodump: write data: upload to S3: RequestError: send request failed
caused by: Post "http://127.0.0.1:9000/bkt/pbm/2020-10-15T17%3A41%3A50Z_rs2.dump.s2?uploads=": dial tcp 127.0.0.1:9000: connect: connection refused.
2020-10-15T20:42:10.000+0300 config/localhost:37028 [ERROR] backup/2020-10-15T17:41:50Z: backup: mongodump: write data: upload to S3: RequestError: send request failed
caused by: Put "http://127.0.0.1:9000/bkt/pbm/2020-10-15T17%3A41%3A50Z_config.dump.s2": dial tcp 127.0.0.1:9000: connect: connection refused.
```
```
$ pbm logs --tail=100 --event=backup/2020-10-15T17:42:54Z
2020-10-15T20:42:54.000+0300 rs2/localhost:28019 [INFO] backup/2020-10-15T17:42:54Z: node in not suitable for backup
2020-10-15T20:43:11.000+0300 rs1/localhost:37020 [INFO] backup/2020-10-15T17:42:54Z: backup started
2020-10-15T20:43:11.000+0300 rs2/localhost:28020 [INFO] backup/2020-10-15T17:42:54Z: backup started
2020-10-15T20:43:11.000+0300 config/localhost:37028 [INFO] backup/2020-10-15T17:42:54Z: backup started
2020-10-15T20:43:11.000+0300 config/localhost:37029 [INFO] backup/2020-10-15T17:42:54Z: backup has been scheduled on another replset node
2020-10-15T20:43:11.000+0300 rs1/localhost:37019 [INFO] backup/2020-10-15T17:42:54Z: backup has been scheduled on another replset node
2020-10-15T20:43:14.000+0300 rs1/localhost:37020 [INFO] backup/2020-10-15T17:42:54Z: s3.uploadPartSize is set to 10485760 (~10Mb)
2020-10-15T20:43:14.000+0300 rs2/localhost:28020 [INFO] backup/2020-10-15T17:42:54Z: s3.uploadPartSize is set to 10485760 (~10Mb)
2020-10-15T20:43:14.000+0300 rs1/localhost:37020 [INFO] backup/2020-10-15T17:42:54Z: mongodump finished, waiting for the oplog
2020-10-15T20:43:14.000+0300 config/localhost:37028 [INFO] backup/2020-10-15T17:42:54Z: s3.uploadPartSize is set to 10485760 (~10Mb)
2020-10-15T20:43:14.000+0300 config/localhost:37028 [INFO] backup/2020-10-15T17:42:54Z: mongodump finished, waiting for the oplog
2020-10-15T20:43:41.000+0300 rs1/localhost:37018 [INFO] backup/2020-10-15T17:42:54Z: backup has been scheduled on another replset node
2020-10-15T20:43:41.000+0300 config/localhost:37027 [INFO] backup/2020-10-15T17:42:54Z: backup has been scheduled on another replset node
2020-10-15T20:43:41.000+0300 rs2/localhost:28018 [INFO] backup/2020-10-15T17:42:54Z: backup has been scheduled on another replset node
2020-10-15T20:43:42.000+0300 rs2/localhost:28020 [INFO] backup/2020-10-15T17:42:54Z: mongodump finished, waiting for the oplog
2020-10-15T20:43:43.000+0300 rs1/localhost:37020 [INFO] backup/2020-10-15T17:42:54Z: s3.uploadPartSize is set to 10485760 (~10Mb)
2020-10-15T20:43:44.000+0300 rs2/localhost:28020 [INFO] backup/2020-10-15T17:42:54Z: s3.uploadPartSize is set to 10485760 (~10Mb)
2020-10-15T20:43:44.000+0300 config/localhost:37028 [INFO] backup/2020-10-15T17:42:54Z: s3.uploadPartSize is set to 10485760 (~10Mb)
2020-10-15T20:43:47.000+0300 config/localhost:37028 [INFO] backup/2020-10-15T17:42:54Z: s3.uploadPartSize is set to 10485760 (~10Mb)
2020-10-15T20:43:48.000+0300 rs1/localhost:37020 [INFO] backup/2020-10-15T17:42:54Z: backup finished
2020-10-15T20:43:48.000+0300 rs2/localhost:28020 [INFO] backup/2020-10-15T17:42:54Z: backup finished
2020-10-15T20:43:48.000+0300 config/localhost:37028 [INFO] backup/2020-10-15T17:42:54Z: backup finished
```

### CODE REVIEW

**Comment by akira-kurogane** [comment]
I see the format -
<utc-timestamp> <severity> <event-name> <msg>.

Looks good to me.

This is similar to MongoDB's pre-v4.4 log format (which was <utc-timestamp> <severity> <component> <thread-name> <msg>).  Severity is D(ebug), I, W, E and F(atal) in increasing severity. For v4.4+ I think "I" is the default minimum verbosity, i.e. you see all except Debug by default, but it was more muddled with an (unprinted) log level before v4.4.

In v4.4 it also changed to a JSON format eg. {"ts": <utc-timestamp>, "s": <severity>, "c": <component>, "ctx": <thread-name>, "msg": (typically shorter, constant string) [, "attrs": { <various if needed> }] }.

Can I suggest we call "--type" "--severity" instead? As for filter in a log filter I think it's typical that we select messages by a >= logic rather than an equality one. E.g. if I select "I"(nfo) it includes W(arning) and E(rror) too?

File: pbm/log/log.go [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+	Tns     int   `bson:"ns" json:"ns"`
+	TZone   int   `bson:"tz" json:"tz"`
+	LogKeys `bson:",inline" json:",inline"`
+	Msg     string `bson:"msg" json:"msg"`
+}
+
+type LogKeys struct {
+	RS      string    `bson:"rs" json:"rs"`
+	Node    string    `bson:"node" json:"node"`
+	Type    EntryType `bson:"type" json:"type"`
```
Comment (akira-kurogane) [inline pbm/log/log.go:36]: This is the field I suggest be called "severity" instead.

File: cmd/pbm/logs.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/percona/percona-backup-mongodb/pbm"
+	plog "github.com/percona/percona-backup-mongodb/pbm/log"
+)
+
+func logs(cn *pbm.PBM) {
```
Comment (akira-kurogane) [inline cmd/pbm/logs.go:12]: Why "logs"? = log-string?

**Review by akira-kurogane — APPROVED** [review]
I have some suggestions, but nothing that precludes this from being merged first.

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| mongodb 3.6 | failed |
| mongodb 4.0 | failed |
| mongodb 4.2 | failed |

**Pull request merged**